deposit_contract/contracts/validator_registration.v.py

```diff
@@ -1,140 +1,99 @@
 
 Deposit: event({
     pubkey: bytes[48],
     withdrawal_credentials: bytes[32],
     amount: bytes[8],
     signature: bytes[96],
-    merkle_tree_index: bytes[8],
 })
-Eth2Genesis: event({deposit_root: bytes32, deposit_count: bytes[8], time: bytes[8]})
 
-zerohashes: bytes32[DEPOSIT_CONTRACT_TREE_DEPTH]
 branch: bytes32[DEPOSIT_CONTRACT_TREE_DEPTH]
 deposit_count: uint256
-full_deposit_count: uint256
-chainStarted: public(bool)
-
 
+# Compute hashes in empty sparse Merkle tree
+zero_hashes: bytes32[DEPOSIT_CONTRACT_TREE_DEPTH]
 @public
 def __init__():
     for i in range(DEPOSIT_CONTRACT_TREE_DEPTH - 1):
-        self.zerohashes[i+1] = sha256(concat(self.zerohashes[i], self.zerohashes[i]))
+        self.zero_hashes[i + 1] = sha256(concat(self.zero_hashes[i], self.zero_hashes[i]))
 
 
 @public
 @constant
 def to_little_endian_64(value: uint256) -> bytes[8]:
-    assert value <= MAX_64_BIT_VALUE
-
-    # array access for bytes[] not currently supported in vyper so
-    # reversing bytes using bitwise uint256 manipulations
+    # Reversing bytes using bitwise uint256 manipulations
+    # (array accesses of bytes[] are not currently supported in Vyper)
     y: uint256 = 0
     x: uint256 = value
-    for i in range(8):
+    for _ in range(8):
         y = shift(y, 8)
         y = y + bitwise_and(x, 255)
         x = shift(x, -8)
-
     return slice(convert(y, bytes32), start=24, len=8)
 
 
 @public
 @constant
 def get_deposit_root() -> bytes32:
-    root: bytes32 = 0x0000000000000000000000000000000000000000000000000000000000000000
+    node: bytes32 = 0x0000000000000000000000000000000000000000000000000000000000000000
     size: uint256 = self.deposit_count
-    for h in range(DEPOSIT_CONTRACT_TREE_DEPTH):
-        if bitwise_and(size, 1) == 1:
-            root = sha256(concat(self.branch[h], root))
+    for height in range(DEPOSIT_CONTRACT_TREE_DEPTH):
+        if bitwise_and(size, 1) == 1:  # More gas efficient than `size % 2 == 1`
+            node = sha256(concat(self.branch[height], node))
         else:
-            root = sha256(concat(root, self.zerohashes[h]))
+            node = sha256(concat(node, self.zero_hashes[height]))
         size /= 2
-    return root
+    return node
+
 
 @public
 @constant
 def get_deposit_count() -> bytes[8]:
     return self.to_little_endian_64(self.deposit_count)
 
+
 @payable
 @public
 def deposit(pubkey: bytes[PUBKEY_LENGTH],
             withdrawal_credentials: bytes[WITHDRAWAL_CREDENTIALS_LENGTH],
             signature: bytes[SIGNATURE_LENGTH]):
-    # Prevent edge case in computing `self.branch` when `self.deposit_count == MAX_DEPOSIT_COUNT`
-    # NOTE: reaching this point with the constants as currently defined is impossible due to the
-    # uni-directional nature of transfers from eth1 to eth2 and the total ether supply (< 130M).
-    assert self.deposit_count < MAX_DEPOSIT_COUNT
+    # Avoid overflowing the Merkle tree
+    assert self.deposit_count < 2**DEPOSIT_CONTRACT_TREE_DEPTH - 1
```

Comment:
At the time of creating `MAX_DEPOSIT_COUNT`, having this constant calculation here was not optimized by the compiler and we took a noticeable (but probably still relatively small) gas hit to calculate dynamically.

Reply:
This actually prevents not only overflowing the merkle tree but prevents the last write which due to the incremental merkle root alg creates a root in error.

I think the comment should reflect this fact

Reply:
How about my second comment here